src/main/java/com/jalasoft/jfc/model/image/ImageConverter.java

```diff
@@ -46,18 +49,25 @@ public FileResult convert(Param param) throws ConvertException, IOException {
         String commandString;
 
         if (imageParam.isThumbnail()) {
-            generateImage(imageParam);
             generateThumbnail(imageParam);
-        } else {
-            generateImage(imageParam);
         }
 
-        ContextStrategy comContext = new ContextStrategy(commandStrategyList);
+        generateImage(imageParam);
+
+        ContextStrategy commandImageContext = new ContextStrategy(commandImageList);
 
         try {
-            commandString = comContext.buildCommand();
+            commandString = commandImageContext.buildCommand();
 
             Runtime.getRuntime().exec(commandString);
+
+            if (!commandThumbnailList.isEmpty()) {
+                ContextStrategy commandThumbnailContext = new ContextStrategy(commandThumbnailList);
+                commandString = commandThumbnailContext.buildCommand();
+
+                Runtime.getRuntime().exec(commandString);
+            }
+
             fileResult = new FileResult();
             fileResult.setPath(imageParam.getOutputPathFile());
         } catch (Exception e) {
```

Comment:
Here would not be one of our exceptions

Reply:
It is in code line 74

Reply:
Name our exceptions.. Exception is generic

Reply:
Sorry, I couldn't understand you.